### PR TITLE
feat: add test case based on jest

### LINKED ISSUE #21: Add test cases to the project
I have planned to add test cases, based on [jest]. The basic work is now complete in [new branch](https://github.com/typed-mxgraph/typed-mxgraph-demo/tree/test-case) and has been tested in [typed-mxgraph-demo](https://github.com/typed-mxgraph/typed-mxgraph-demo), a simple project. At the same time I modified the project structure in order to distinguish between tests and source. Because of a lot of changes, I can't merge to the master branch hastily, @tbouffard please help me to review.
 
[jest]: https://jestjs.io

### PR DESCRIPTION
**Pull request by hungtcs**
- move source to lib
- add tests folder

Closes #21

### CODE REVIEW

File: tests/factory.spec.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import factory, { mxGraphExportObject } from 'mxgraph';
+
+describe('factory', () => {
+  let mxgraph: mxGraphExportObject;
+
+  beforeAll(() => {
+    mxgraph = factory();
+  });
+
+  it('should be create', () => {
```
Comment (tbouffard) [inline tests/factory.spec.ts]:
```suggestion
  it('should be created', () => {
```

**Comment by tbouffard** [comment]
✔️ Tested on https://github.com/typed-mxgraph/typed-mxgraph-demo with using [npm link](https://docs.npmjs.com/cli/v6/commands/npm-link): 
From the demo repository, create the link with `npm link ../typed-mxgraph`, and check that types are available.
I am still testing other projects that use the types.

**Comment by tbouffard** [comment]
✔️ tested on https://github.com/tbouffard/mxgraph-typescript-example-bundled-with-rollup with npm link successfully.
Notice that this repository uses similar mxGraph classes to those used in the `typed-mxgraph-demo` repository.

**Comment by hungtcs** [comment]
Thanks @tbouffard , combined with our tests, I think it can be merged.

**Comment by tbouffard** [comment]
Very cool! 🎉 

Just to let you know that in the meantime, I am currently working on migrating to `typed-mxgraph` in [bpmn-visualization](https://github.com/process-analytics/bpmn-visualization-js/) and I don't have seen any issue with the latest `1.0.0-1` release including the changes of this Pull Request.
`bpmn-visualization` uses much more mxGraph classes than the 2 examples we have tested, so this is a good indicator.

**Pull request merged**